controllers/storagecluster/reconcile.go

```diff
@@ -159,14 +161,24 @@ func (r *StorageClusterReconciler) Reconcile(ctx context.Context, request reconc
 		return reconcile.Result{}, err
 	}
 
+	var err error
+	availableCrds, err := statusutil.MapCRDAvailability(ctx, r.Client, statusutil.GetCrds()...)
```

Comment:
We are using a cached client and we have the ability to add a custom index to the cache (if needed). Why do we need this functionality anymore?